### PR TITLE
vscode/asdf: search the default XDG base data directory

### LINKED ISSUE #2023: vs code: asdf environment variables ignored
## VS Code Ruby LSP
The extension detects `asdf` but fails when searching for the data directory.

The [code](https://github.com/Shopify/ruby-lsp/blob/main/vscode/src/ruby/asdf.ts) does not appear to use the asdf environment variables:
* `ASDF_DIR`
* `ASDF_DATA_DIR`

Instead, it attempts to detect those values by looking at predetermined locations. This fails when a those values are set to alternative locations. 

### Reproduction steps

1. set `ASDF_DATA_DIR` to ~/asdf-data
2. install/setup asdf to that location (see [docs](https://asdf-vm.com/manage/configuration.html#environment-variables))
3. Open VS Code
4. Open a Ruby file
5. See error:

```
Automatic Ruby environment activation with asdf failed:
Could not find ASDF data dir. Searched in
file:///Users/zach.gianos/.asdf, 
file:///opt/asdf-vm, 
file:///opt/homebrew/opt/asdf/libexec, 
file:///usr/local/opt/asdf/libexec
```

### Note

A recent update broke this customization of the `asdf` setup.

I presume that it was this commit: [30bc0001](https://github.com/Shopify/ruby-lsp/commit/30bc0001b1085463d7bb420756c2de662c34202b).

I'm simply comparing the timeframe of the commit and the functionality change. (I use Ruby LSP every day.)

### PR DESCRIPTION
**Pull request by gi**
This updates the detection of the asdf data directory for setting the environment variable `ASDF_DATA_DIR` during activation.

Fixes #2023

### Motivation

Fixes #2023 

The XDG base directory specification is becoming more popular in outside of Linux. The default base data directory is `~/.local/share` where applications store their user data within a directory under the base: e.g. `~/.local/share/asdf`.

### Implementation

This update includes the XDG base data directory in the list when detecting the location for activation.

### Automated Tests

Yes.

### Manual Tests

1. Create the directory `~/.local/share/asdf/shims`
2. Start VS Code
3. Open Ruby file

### CODE REVIEW

File: vscode/src/ruby/asdf.ts [hunk 1]
```diff
@@ -51,8 +51,19 @@ export class Asdf extends VersionManager {
   // file, but that may not be the case for a Homebrew installation, in which case the we'd have
   // `/opt/homebrew/opt/asdf/libexec/asdf.sh`, but the data directory might be `~/.asdf`
   async findAsdfDataDir(): Promise<vscode.Uri> {
+    // In order, the data locations are:
+    // 1. Default
```
Comment (andyw8) [inline vscode/src/ruby/asdf.ts:55]: (nit) If we assign each possible path to variable, then we could avoid need for this comment.

**Comment by andyw8** [comment]
(the rest of the team is at RubyKaigi so there might be a little delay until we ship this)

**Comment by vinistock** [comment]
@gi I shipped a few improvements to the ASDF implementation which should fix many of the issues we were seeing before. I think this PR can just add the `.local/share` directory to the list and that should be good 👍.

**Comment by gi** [comment]
@vinistock I just tested the latest version, and the changes work on my machine. Thanks for updating. I'm going to close this as the changes are no longer needed.

The issue was fixed by https://github.com/Shopify/ruby-lsp/pull/2006

**Pull request closed without merging**